lib/internal/http2/util.js

```diff
@@ -350,6 +354,67 @@ function getSettings(session, remote) {
 
 function updateSettingsBuffer(settings) {
   let flags = 0;
+  let numCustomSettings = 0;
+
+  if (typeof settings.customSettings === 'object') {
+    const customSettings = settings.customSettings;
+    for (const setting in customSettings) {
+      const val = customSettings[setting];
+      if (typeof val === 'number') {
+        let set = false;
+        const nsetting = Number(setting);
```

Comment:
This should verify that `nsetting` is not `NaN`

Reply:
Included in check a bit more above.